.github/workflows/deploy.yml

```diff
@@ -0,0 +1,53 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - develop
+      - master
+
+jobs:
+  canary:
+    runs-on: ubuntu-latest
+    if: github.ref_name == 'develop'
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+      # Make sure the working tree is clean
+      - run: git reset --hard HEAD
+      - run: |
+          npx lerna publish prepatch \
+            --no-verify-access \
+            --no-git-tag-version \
+            --no-push \
+            --canary \
+            --dist-tag=next \
+            --exact \
+            --force-publish \
+            --yes
+
+  production:
+    runs-on: ubuntu-latest
+    if: github.ref_name == 'master'
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+      # Make sure the working tree is clean
+      - run: git reset --hard HEAD
+      - run: |
+          npx lerna publish from-package \
+            --force-publish='*' \
+            --yes
+      - uses: softprops/action-gh-release@v1
```

Comment:
Tags are mutable, so this sort of `uses` directive amounts to using an unpinned dependency version in an environment containing a publishing secret. Safer to pin to a commit hash rather than the `v1` tag.

Reply:
While it's safer, it's also a maintenance nightmare ☹️

Reply:
Renovate supports a syntax to [use digest pinning but also understand to update according to tags](https://docs.renovatebot.com/modules/manager/github-actions/#additional-information), which is the best of both worlds. That's what I recommend we end up with.

Reply:
Dependabot [also appears to support this](https://github.com/dependabot/dependabot-core/pull/5951), though I didn't see documentation about it.

Reply:
I'll pin this action to a commit, but am worried that unless we get Renovate setup, using actions this way is going to be a pain.